### DISCUSSION TITLE
Deleting an item by id causes the last item added to the db to be deleted instead

### DISCUSSION THREAD

**Discussion by AvivShvitzky** [prisma/prisma#17896] [Q&A] [answered]
I'm using cockroachDB and I have this code which should delete a note item by its id:

```
const deleteNote = async (id) => {
  return await prisma.note.delete({
    where: {
      id: id,
    },
  });
};
```

What happens instead is that the last note added to the db is deleted.
I have confirmed that a correct id is being sent each time.

This is odd because I've already got deletion working for other tables the same way.

---

further info:
**Schema**:
```
model Note {
  id        String    @id @default(cuid())
  title     String?
  content   String
  createdAt DateTime  @default(now())
  updatedAt DateTime? @updatedAt
  story     Story     @relation(fields: [storyId], references: [id])
  storyId   String
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @AvivShvitzky 👋 

It might be useful to log the notes that match the given filter criteria before performing the deletion to see what's happening. You can do this by using the .findMany method in Prisma Client:

```js
const deleteNote = async (id) => {
  const note = await prisma.note.findUnique({
    where: {
      id: id,
    },
  });

  console.log(note);

  return await prisma.note.delete({
    where: {
      id: id,
    },
  });
};
```
This should give you a better understanding of what's happening and help you identify the root cause of the issue. Also, to ensure that the correct note is deleted, you can specify additional filter criteria in the where argument. For example, you could include a unique column in the where argument to ensure that only one specific note is deleted.

**Reply by nikolasburk** [reply]
That's a good way to start debugging this! 

You can also add more [logging and debugging output](https://www.prisma.io/docs/guides/database/troubleshooting-orm/creating-bug-reports#include-logging-and-debugging-output) to understand what exactly Prisma is doing under the hood.

**Reply by AvivShvitzky** [reply]
Thank you guys for your comments.
It turns out the right notes have been deleted, but I have a UI problem that makes it look like the wrong ones are deleted. 
THE ONE TIME I DON'T DOUBLE-CHECK ON PRISMA CLIENT.
Sorry, you guys!

**Reply by ludralph** [reply]
No worries @AvivShvitzky. We are glad to be of help.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
